Azure pipeline that uses the CodeExporterApp to convert samples & make PRs to destination git repos

turbobobbytraykov opened the pull request.

turbobobbytraykov committed:
Initial draft of the pipeline

turbobobbytraykov committed:
Set up CI with Azure Pipelines

[skip ci]

turbobobbytraykov committed:
Fix typo in JSON file

turbobobbytraykov committed:
Tweak to the Powershell script

turbobobbytraykov committed:
Get artifact from code-generation-library repo's pipeline

turbobobbytraykov committed:
Checkout igniteui-angular-examples

turbobobbytraykov committed:
Hardcode vnext branch for secondary repo

turbobobbytraykov committed:
Merge branch 'azure-pipeline-use-code-exporter-app' of https://github.com/IgniteUI/igniteui-xplat-examples into azure-pipeline-use-code-exporter-app

turbobobbytraykov committed:
Update build-pipeline.yml for Azure Pipelines

turbobobbytraykov committed:
Update build-pipeline.yml for Azure Pipelines

turbobobbytraykov committed:
Angular-related modifications

turbobobbytraykov committed:
Path fixes in the build pipeline

turbobobbytraykov committed:
Fix typo in CodeExporerApp invokation

turbobobbytraykov committed:
Fix path to CodeExporterApp

turbobobbytraykov committed:
Put everything in $(Build.SourcesDirectory)

turbobobbytraykov committed:
Add all other examples repos and their corresponding invokations of the CodeExporerApp

turbobobbytraykov committed:
Fix resources section

turbobobbytraykov committed:
Add display names to the steps that invoke the CodeExporterApp

turbobobbytraykov committed:
Temporarily remove Powershell json parsing - need to gauge performance dip/added value

turbobobbytraykov committed:
Use the igeditor-templates and code-generation-libraries as git repos instead of pipeline artifacts

turbobobbytraykov committed:
Update build-pipeline.yml for Azure Pipelines

turbobobbytraykov committed:
Omit powershell core installation since it doesn't work in a concurrent environment

turbobobbytraykov committed:
Experiment with more than one stage

turbobobbytraykov committed:
Use same agent pool for all stages

turbobobbytraykov committed:
Update build-pipeline.yml for Azure Pipelines

turbobobbytraykov committed:
Update build-pipeline.yml for Azure Pipelines

turbobobbytraykov committed:
Jesus Christ, this shortcutting of pipeline steps is infuriating

turbobobbytraykov committed:
Use JSON sample files from the Prepare stage instead

turbobobbytraykov committed:
Fixing the path to the downloaded XPlatExamples from the previous stage

turbobobbytraykov committed:
Explicitly forbid this repo's cloning in the Build stage - we get these files from the previous stage

turbobobbytraykov committed:
Split convertion operations into separate jobs

turbobobbytraykov committed:
Tweaks to the pipeline

turbobobbytraykov committed:
try using lower caps for the pipeline artifact's name

turbobobbytraykov committed:
Update build-pipeline.yml for Azure Pipelines

turbobobbytraykov committed:
Add missing coma

turbobobbytraykov committed:
Update build-pipeline.yml for Azure Pipelines

turbobobbytraykov committed:
Add aggressive workspace cleaning

turbobobbytraykov committed:
Update build-pipeline.yml for Azure Pipelines

turbobobbytraykov committed:
Update build-pipeline.yml for Azure Pipelines

turbobobbytraykov committed:
Fix path to input JSONS for Blazor

turbobobbytraykov committed:
Explicitly install jq on agents

turbobobbytraykov committed:
Try to use Invoke-Expression for the json file purging script

turbobobbytraykov committed:
Pipeline triggers-related changes - fine tuning

turbobobbytraykov committed:
Experimental changes to the pipeline

turbobobbytraykov committed:
Pipeline experiments: get changed files

turbobobbytraykov committed:
Pipeline tweaking

turbobobbytraykov committed:
Need to fail the stage if the call to the CodeExporter.App.exe reports any errors

turbobobbytraykov committed:
Fix error handling of Invoke-Expression

turbobobbytraykov committed:
Fix typo in pipeline

turbobobbytraykov committed:
Invoke-Expression fixes in the pipeline

turbobobbytraykov committed:
Revert experiments with Invoke-Expression - no longer relevant to the implemenetation

turbobobbytraykov committed:
Update build-pipeline.yml for Azure Pipelines

turbobobbytraykov committed:
Tweaks to finding what files were changed

turbobobbytraykov committed:
Temp reductions of samples being parsed

turbobobbytraykov committed:
Temp comment out all platforms except Angular

turbobobbytraykov committed:
Update build-pipeline.yml for Azure Pipelines

turbobobbytraykov committed:
Update build-pipeline.yml for Azure Pipelines

turbobobbytraykov committed:
Pipeline: store changed samples for dev troubleshooting

turbobobbytraykov committed:
Add conditional PR making to the destination examples repo (Angular)

turbobobbytraykov committed:
Remove silly copy-pasted code

turbobobbytraykov committed:
PR-related tweaks to the pipeline

turbobobbytraykov committed:
First attempt at a template for the pipeline's jobs

turbobobbytraykov committed:
Add reference to template in pipeline

turbobobbytraykov committed:
Remove experimental PR-related functionality

turbobobbytraykov committed:
Fix YAML identation

turbobobbytraykov committed:
Fix YAML identation again

turbobobbytraykov committed:
Use lowercase platform name in pipeline

turbobobbytraykov committed:
Fix pipeline template

turbobobbytraykov committed:
Fix pipeline template

turbobobbytraykov committed:
Expose a few other variables as template parameteres

turbobobbytraykov committed:
Fix typo in the pipeline

turbobobbytraykov committed:
Use template for all platforms in the pipeline

turbobobbytraykov committed:
Use parameter for job name inside the template

turbobobbytraykov committed:
Further work on the pipeline & its job template

turbobobbytraykov committed:
Camel case for React as a platform

turbobobbytraykov committed:
Improve reason for making a PR on the destination git repos

turbobobbytraykov committed:
Move pathsToParse to be an input parameter instead of a variable

turbobobbytraykov committed:
pathsToUse is now a subpath in the template

turbobobbytraykov committed:
Fix typo in input param's property

turbobobbytraykov committed:
Tweaking the pipeline template

turbobobbytraykov committed:
Fix typo in the template

turbobobbytraykov committed:
Fix parameter for destination git repo in the pipeline template

turbobobbytraykov committed:
Fix input parameter reference when calling the pipeline template

turbobobbytraykov committed:
Troubleshooting PR creation error

turbobobbytraykov committed:
Tweak pathsToParse input parameter

turbobobbytraykov committed:
Create explicit dummy change - troubleshooting

turbobobbytraykov committed:
PR creation troubleshooting

turbobobbytraykov committed:
Try making a PR to docfx - experimental RP troubleshooting

turbobobbytraykov committed:
Change service conn for igniteui-docfx repo cloning

turbobobbytraykov committed:
PWD in temp PS script

turbobobbytraykov committed:
Add code to make the PR's source branch - experimental stuff

turbobobbytraykov committed:
Fix typo in condition

turbobobbytraykov committed:
Update build-pipeline.yml for Azure Pipelines

turbobobbytraykov committed:
Update build-pipeline.yml for Azure Pipelines

turbobobbytraykov committed:
Update build-pipeline.yml for Azure Pipelines

turbobobbytraykov committed:
Work on the pipeline template

turbobobbytraykov committed:
Comment out PR-making experiment

turbobobbytraykov committed:
Work on the pipeline template

turbobobbytraykov committed:
Merge branch 'azure-pipeline-use-code-exporter-app' of https://github.com/IgniteUI/igniteui-xplat-examples into azure-pipeline-use-code-exporter-app

turbobobbytraykov committed:
Work on the pipeline template

turbobobbytraykov committed:
Comment out more experimental code

turbobobbytraykov committed:
Work on the pipeline template

turbobobbytraykov committed:
Work on the pipeline template - temp print messages

turbobobbytraykov committed:
Work on the pipeline template - tweak git command handling

turbobobbytraykov committed:
Work on the pipeline template - tweak git command handling

turbobobbytraykov committed:
Remove commented-out code from the pipeline

turbobobbytraykov committed:
Update PR title - more details

turbobobbytraykov committed:
Fix location of pre-processed JSON files

HUSSAR-mtrela commented on azure-pipelines/build-pipeline.yml:
```diff
@@ -0,0 +1,148 @@
+      endpoint: IgniteUI
+      type: github
+      name: IgniteUI/igniteui-angular-examples
+      ref: '$(Build.SourceBranch)'
```
for now, exclude exporting this repo to IgniteUIAngularExamples because our Angular template is not complete

HUSSAR-mtrela commented on azure-pipelines/build-pipeline.yml:
```diff
@@ -0,0 +1,148 @@
+      platform: 'Angular'
+      destinationGitRepo: 'igniteui-angular-examples'
+      pathsToParse: ${{ parameters.pathsToParse }}
+      codeExporterAppBranch: $(codeExporterAppBranch)
```
for now, exclude exporting this repo to IgniteUIAngularExamples because our Angular template is not complete

HUSSAR-mtrela commented:
@turbobobbytraykov see my comment above

turbobobbytraykov committed:
Tweak path to processed JSON files

turbobobbytraykov committed:
Tweaks to the pipeline & its template

turbobobbytraykov committed:
Merge branch 'azure-pipeline-use-code-exporter-app' of https://github.com/IgniteUI/igniteui-xplat-examples into azure-pipeline-use-code-exporter-app

turbobobbytraykov committed:
Exclude Angular processing since their template is not complete

turbobobbytraykov committed:
Exclude Angular processing since their template is not complete

turbobobbytraykov committed:
Expose new parameter to the build pipeline's template - to be used by a 2nd pipeline soon. Also make sure that this pipeline will start making PRs to the vnext branch

turbobobbytraykov committed:
Adjust implementation in pipeline template

turbobobbytraykov committed:
Remove reliance on xplatexamples build artifact - no longer relevant

turbobobbytraykov committed:
Fix typo in the pipeline template

turbobobbytraykov committed:
Fix path to JSONs - as an input param value in the pipeline

turbobobbytraykov committed:
Tweaks to the build pipeline
Added a new pipeline for the devs to just run the CodeExporter tool against a feature branch of this repo

turbobobbytraykov committed:
Clarification on which branch of the CodeExporer we are using in the pipelines

turbobobbytraykov committed:
Target vnext as the destination branch for RPs that this pipeline creates - expected behavior

turbobobbytraykov committed:
Explicitly target the vnext branch of the destination git repo when doing the PR

turbobobbytraykov committed:
Alter title of created PRs

turbobobbytraykov committed:
Alter title and description of created PRs

turbobobbytraykov committed:
Alter title and description of created PRs

HUSSAR-mtrela commented on azure-pipelines/dev-run-pipeline.yml:
```diff
@@ -0,0 +1,98 @@
+      endpoint: IgniteUI
+      type: github
+      name: IgniteUI/igniteui-wc-examples
+      #ref: vnext
```
@turbobobbytraykov I think this is a mistake to commente out vnext branch, right?

turbobobbytraykov replied:
@HUSSAR-mtrela , it was unintentional - fixed.

turbobobbytraykov committed:
Explicitly checkout the vnext branch in the pipeline that the devs can use to examine their changes

turbobobbytraykov committed:
Add a tag to the generated PR - requested by Martin Trela
